robot_tests/robot_tests/test_executor.py

```diff
@@ -0,0 +1,13 @@
+import os
+from robot import run
+package_directory = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestExecutor:
+    def __init__(self, host: str, port: int):
+        self.host = host
+        self.port = port
+
+    def execute_tests(self, include_tags=[], exclude_tags=[], output_dir="robot_results"):
+        run(f"{package_directory}/tests", include=include_tags,exclude=exclude_tags, outputdir=output_dir,
```

Comment:
no need to do this now, but is it possible for us to get test results/other data back from this method? It would make it very easy to store info in the utf `state` to give it to the reporter step

Reply:
Yeah, I had the same thought; curious if Tanner has any ideas.  I'll see if anything pops into mind once I read through the rest of the PR as well (need to see how some things are structured).

Reply:
Good idea. I think we could pretty easily store the location of the output files (Which should be placed in the UTF workspace dir as Chris mentioned) from these tests in the `utf` state (Not necessarily as a return of this function) and then output this in the reporter step. I think this is all I had in mind at this point but open to other thoughts